### DISCUSSION TITLE
Node replacement

### DISCUSSION THREAD

**Discussion by yinhen** [minio/minio#14335] [Q&A] [answered]
we found that it's not convenient to replace the node in minio cluster if one of the nodes or drivers failed. So I want to know If we have any plan to solve this problem. I think it's a basic ability to replace node in distributed storage system, minio would be more popular if minio support using mc command to replace node

**Comment by klauspost** [comment, marked as answer]
Drive replacement is https://docs.min.io/minio/baremetal/installation/restore-minio.html#drive-failure-recovery

1) Unmount the failed drive(s)
2) Replace the failed drive(s)
3) Review and Update fstab
4) Remount the Replaced Drive(s)

It doesn't get much easier than that.

Replacing a node https://docs.min.io/minio/baremetal/installation/restore-minio.html#node-failure-recovery

1) Start the Replacement Node
2) Update Hostname for the New Node
3) Download and Prepare the MinIO Server
4) Rejoin the node to the deployment

Again, I would like to know what could be made easier.

But if you require more detailed assistance that is in the docs, look into our [subscription offer](https://min.io/pricing).

**Reply by yinhen** [reply]
What I want to argue is the Step2 in replacing a node. If replacement node has a different IP address from the failed host，it's not convenient to tackle this problem. What I hope is that in the future，we could use mc command to replace a old node by a new one with different IP and hostname, new node comes online and old one goes offline.Nothing changes except one node in the cluster.

**Reply by harshavardhana** [reply]
just keep the same hostname - how hard is that? @yinhen - it is unnecessary and unneeded complexity in deployment to have some random `mc` commands to replace node names.  Use hostnames always when you don't control the network fabric.
